### PR TITLE
Bug 1856814 - add namespace list  to job pod as an env var

### PR DESCRIPTION
**Pull request by jmontleon**
https://bugzilla.redhat.com/show_bug.cgi?id=1856814

### CODE REVIEW

**Comment by jmontleon** [comment]
I don't know if anyone has ideas on what other information might be useful to add besides these.

**Comment by eriknelson** [comment]
+1 from me, although I dunno if I should be the one acking this through.

**Comment by djwhatle** [comment]
I remember thinking this type of information would be really useful to have in the hook runner container. +1, ACK

**Comment by djwhatle** [comment]
\test

**Comment by djwhatle** [comment]
Wow that CI finished way too fast. Anyways, simple change, so merge away.

File: pkg/controller/migmigration/hooks.go [hunk 1]
```diff
@@ -235,6 +235,16 @@ func (t *Task) baseJobTemplate(hook migapi.MigPlanHook, migHook migapi.MigHook)
...
 							Name:  strings.ToLower(t.PlanResources.MigPlan.Name + "-" + hook.Phase),
 							Image: migHook.Spec.Image,
+							Env: []corev1.EnvVar{
+								{
+									Name:  "MIGRATION_NAMESPACES",
+									Value: strings.Join(t.PlanResources.MigPlan.Spec.Namespaces, ","),
+								},
+								{
+									Name:  "MIGRATION_NAME",
+									Value: t.PlanResources.MigPlan.Name,
```
Comment (djwhatle) [inline pkg/controller/migmigration/hooks.go:245]: @jmontleon this looks wrong. Shouldn't this be `meta.name`?

Comment (djwhatle) [reply pkg/controller/migmigration/hooks.go:245]: This is also the `MIGRATION_PLAN_NAME`, not the `MIGRATION_NAME`. Maybe we should update to reflect the data we're providing.

Comment (jmontleon) [reply pkg/controller/migmigration/hooks.go:245]: It works correctly. It's also how the hook name has been generated all along:
https://github.com/konveyor/mig-controller/blob/master/pkg/controller/migmigration/hooks.go#L170

```
$ oc get po hooktest-prebackup-8hjzx-znlq7 -o yaml | grep MIGRATION -A1
    - name: MIGRATION_NAMESPACES
      value: registry-images
    - name: MIGRATION_NAME
      value: hooktest

$ oc get migplan
NAME       AGE
hooktest   26h
```

Not objection to changing the env var name.

Comment (jmontleon) [reply pkg/controller/migmigration/hooks.go:245]: unless I'm doing it wrong and misunderstanding Meta.Name doesn't work
image

**Comment by jmontleon** [comment]
Updated `MIGRATION_NAME` to `MIGRATION_PLAN_NAME`

File: pkg/controller/migmigration/hooks.go (hunk 1 above)
Comment (djwhatle) [reply pkg/controller/migmigration/hooks.go:245]: Ahh, my bad. I forgot realize TypeMeta and ObjectMeta and inlined on the MigPlan struct so they're available from top-level like that. Spec isn't inlined.

```
type MigPlan struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`
```

**Pull request merged**